src/sentry/static/sentry/app/views/settings/components/forms/rangeField.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import InputField from 'app/views/settings/components/forms/inputField';
+import RangeSlider from 'app/views/settings/components/forms/controls/rangeSlider';
+
+type DefaultProps = {
+  formatMessageValue?: false | Function;
+};
+
+type DisabledFunction = (props: any) => boolean;
+type PlaceholderFunction = () => string;
+
+type Props = DefaultProps &
+  Omit<RangeSlider['props'], 'value' | 'disabled' | 'placeholder'> &
+  Omit<InputField['props'], 'field'> & {
+    disabled?: boolean | DisabledFunction;
+    placeholder?: string | PlaceholderFunction;
+  };
+
+function onChange(fieldOnChange, value, e) {
+  fieldOnChange(value, e);
+}
+
+function defaultFormatMessageValue(value, props: Props) {
```

Comment:
Is the value of type string here?

Reply:
`formatLabel` in this instance expects a number, so no. It can also be `false` I believe, which acts as a flag to change behaviour. I've left a lot of these functions more loosely typed for now as these props need to conform to to the generic `fieldFromConfig.tsx`, which makes inferring usage harder.